executor/executor_runner.h

```diff
@@ -215,7 +262,13 @@ class Proc
 				while (ReadOutput())
 					;
 			}
-			HandleCompletion(status);
+			bool hanged = SYZ_EXECUTOR_USES_FORK_SERVER && state_ == State::Executing;
+			HandleCompletion(status, hanged);
+			if (hanged) {
+				// If the process has hanged, it may still be using per-proc resources,
+				// so allocate a fresh proc id.
+				id_ = proc_id_pool_.Alloc(id_);
```

Comment:
Nit: let's maybe `debug()` print the change of the pid? Just to make the debugging logs less surprising.

Reply:
Done